### DISCUSSION TITLE
Unable to retrieve the current version certificate from CSI drivers, its still pointing to older version

### DISCUSSION THREAD

**Discussion by KunalAdu** [Azure/secrets-store-csi-driver-provider-azure#623] [Q&A]
Installed CSI drivers and able to fetch the certificate from the Key vault , Uploaded a new version of certificate as older version is going to expire now in key vault there are two version i.e. current version and Older version if i delete the kubernetes secret it is creating new secret in approx. 5 mints but it is pointing to older version , how to refresh that and point it to new version automatically ?

What did I expect to happen:
It should retrieve the current version as secret provider is pointing to keyvault

**Comment by aramase** [comment]
> It should retrieve the current version as secret provider is pointing to keyvault

You can enable secret auto rotation to periodically check for new versions and update the secrets/keys/certs in the pod volume: https://azure.github.io/secrets-store-csi-driver-provider-azure/configurations/sync-with-k8s-secrets/.
